ibis/backends/mssql/__init__.py

```diff
@@ -76,6 +76,43 @@ def do_connect(
         driver: str | None = None,
         **kwargs: Any,
     ) -> None:
+        """Connect to MSSQL database.
+
+        Parameters
+        ----------
+        host
+            Address of MSSQL server to connect to.
+        user
+            Username.  Leave blank to use Integrated Authentication.
+        password
+            Password.  Leave blank to use Integrated Authentication.
+        port
+            Port of MSSQL server to connect to.
+        database
+            The MSSQL database to connect to.
+        driver
+            ODBC Driver to use.
+
+            On Mac and Linux this is usually 'FreeTDS'.
+
+            On Windows, it is usually one of:
+
+            - ODBC Driver 11 for SQL Server
+            - ODBC Driver 13 for SQL Server (for both 13 and 13.1)
+            - ODBC Driver 17 for SQL Server
+            - ODBC Driver 18 for SQL Server
+
+            See https://learn.microsoft.com/en-us/sql/connect/odbc/windows/system-requirements-installation-and-driver-files
+        kwargs
+            Additional keyword arguments to pass to PyODBC.
+        """
+
+        # If no user/password given, assume Windows Integrated Authentication
+        # and set "Trusted_Connection" accordingly
+        # see: https://learn.microsoft.com/en-us/sql/connect/odbc/linux-mac/using-integrated-authentication
+        if user is None and password is None and "Trusted_Connection" not in kwargs:
+            kwargs["Trusted_Connection"] = "yes"
```

Comment:
Could go either way on this, just commenting show a possibly cleaner spelling:

```suggestion
        if user is None and password is None:
            kwargs.setdefault("Trusted_Connection", "yes")
```